### DISCUSSION TITLE
incremental rotary encoders for moving the Z and X axes in manual mode

### DISCUSSION THREAD

**Discussion by sq2eer** [kachurovskiy/nanoels#117] [Ideas]
Hello Maxim, I put photos of how I have the headstock encoder installed in my lathe and what encoders I have to move the Z and X axes in manual mode and photos if I could see it, the question is whether it is possible to connect to your controller

![20230524_200842](https://github.com/kachurovskiy/nanoels/assets/120673557/531231d0-be53-4851-aeb3-6f0584e90bc9)

![20230524_200859](https://github.com/kachurovskiy/nanoels/assets/120673557/b6aa3f78-2480-4bc3-9404-bde7f811ca08)

![20230524_200912](https://github.com/kachurovskiy/nanoels/assets/120673557/ec08f8a1-6b70-4be9-b36a-a77c34ae922c)

![20230524_200918](https://github.com/kachurovskiy/nanoels/assets/120673557/3b00e61d-1886-43a2-8020-d95aaecdcf37)

greetings Jacek.

**Comment by kachurovskiy** [comment]
Hi Jacek, interesting idea. Encoder you have on the main spindle appear to be the right one. For the axes, usually a different type is used: https://www.aliexpress.com/item/1005004437852504.html

Reason is that your encoders have no holding power and will easily be disturbed by vibrations, especially if the handle you put on them is weight-imbalanced (like the one on the last picture).

Pin-wise there's little difference between https://www.aliexpress.com/item/1005004437852504.html and what you have except PPR (pulses per revolution) - optical encoders usually have 300-1200 pulses while handwheels have 100-200 which is more suitable for this application.

Let me know what you think. Cheers!

**Comment by sq2eer** [comment]
Yes, I know that the encoder with the knob alone will not pass the exam, so I plan to make an adjustable brake with felt (I do not know if the translator translated the word "with felt" well)  but if your controller can handle the 720 imp/rev encoder in the spindle and 1000 imp/rev rpm in the "knobs" of the Z and X axes

**Reply by kachurovskiy** [reply]
Sure, ESP32-S3 can handle that.

Question is whether you think it's still safe to operate the lathe in this manner - doing 700 pulses with one hand motion will send your cross-slide 70cm away at 1mm step and you might not have intended that. You could stop it by pressing ![IconStop](https://github.com/kachurovskiy/nanoels/assets/517919/7c472f0c-8b5c-4ab3-991f-7d7782447d03) but by the time you reach it the damage could already be done.

**Comment by sq2eer** [comment]
I thought that in manual mode, 1 turn of the knob corresponds to 1 turn of the lead screw
at 1000 imp/rev this would give high resolution.
For me, the Z-axis feed screw has a pitch of 3mm/rev, it is a trapezoidal screw, I was thinking of replacing it with a ball screw with a pitch of 5mm/rev
and in the X axis I have 2mm/rev and I would like to replace it with a ball with a pitch of 4mm/rev, of course, it could be rounded to 2 decimal places, i.e. to 0.03 mm for the Z axis and 0.02 mm for the X axis before replacing the screws. What do you think.

**Reply by kachurovskiy** [reply]
That's an interesting take, indeed if we model those pulse generators as handles for the actual lead screws things get safer and more natural.

How should it work when Z is being moved and ELS gearbox is ON? Z is supposed to stay in the thread at the end of the move.

**Comment by sq2eer** [comment]
I thought to just replace the axis shift buttons with encoders but without setting the feed pitch, only 1 encoder turn 1 screw turn or add buttons or encoder as a switching option

**Comment by sq2eer** [comment]
hi, once I tried to build such a driver myself, I even designed a printed circuit board, but nothing came of it, I'm too weak to write programs, my driver is still with lcd DWIN https://www.youtube.com/watch?v=VKvhEHLMDHk
good night until tomorrow.

**Reply by kachurovskiy** [reply]
Here's how I have it implemented provisionally - https://photos.app.goo.gl/4Mmwj32RNvKGZkbV9 - I don't happen to have spare optical encoders but they have the same wiring I think.

**Comment by sq2eer** [comment]
Hello Maxim, this is what I would like to have, I understand that it will be possible to connect 2 encoders, one for the Z axis and one for the X axis, my encoders have A, B, and Z signals so they will fit I have to finish another project   https://www.youtube.com/watch?v=2pTekg34ozYand I will buy your driver to modify my lathe
Regards Jacek.

**Reply by kachurovskiy** [reply]
There's not enough pins available to connect the Z signal for encoders, just A and B. I'm not selling in EU yet so no hurry :) Nice project.
